x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportForecastJobAction.java

```diff
@@ -103,16 +113,33 @@ protected void taskOperation(ForecastJobAction.Request request, JobTask task, Ac
                     }
 
                     ForecastParams params = paramsBuilder.build();
-                    processManager.forecastJob(task, params, e -> {
+                    ActionListener<Boolean> resultsMappingUpdateHandler = ActionListener.wrap(
+                    r -> processManager.forecastJob(task, params, e -> {
                         if (e == null) {
                             getForecastRequestStats(request.getJobId(), params.getForecastId(), listener);
                         } else {
                             listener.onFailure(e);
                         }
+                    }),
+                    ex -> {
+                        logger.warn(
+                            new ParameterizedMessage(
+                                "[{}] job failed to update results mappings before requesting forecast",
+                                request.getJobId()
+                            ),
+                            ex);
+                        listener.onFailure(ex);
                     });
+                    ElasticsearchMappings.addDocMappingIfMissing(
```

Comment:
I think the changes in this file are unnecessary, because from the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-forecast.html):

> The job must be open when you create a forecast. Otherwise, an error occurs.

And we already update the mappings when opening a job: https://github.com/elastic/elasticsearch/blob/034926651708cbe8e96f28d10e029e1f3bf1a3a9/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectProcessManager.java#L484-L495

Reply:
@droberts195 🤔 ah and the node version cannot change while the job is opened :D. 🤦